versioned_docs/version-4.0.0-alpha.1/datastructures/custom.md

```diff
@@ -3,48 +3,116 @@ title: Custom Data Structures
 slug: /datastructures/custom-datastructure
 ---
 
-:::caution
-TODO
+While the `ink_storage` crate provides useful utilities and data structures to organize and 
+manipulate the contract's storagem, contract authors are not limited by its capabilities. 
 
-Beware, this page is no longer up to date for 4.0!
-:::
+## Using custom types on storage
```

Comment:
Perhaps we could merge the "Pre-ink! v4.0.0-beta storage" chapter here, to provide information about how new traits differ from the old traits?